workflow/controller/taskresult.go

```diff
@@ -54,19 +54,14 @@ func (wfc *WorkflowController) newWorkflowTaskResultInformer() cache.SharedIndex
 	return informer
 }
 
-func podAbsentTimeout(node *wfv1.NodeStatus) bool {
-	return time.Since(node.StartedAt.Time) <= envutil.LookupEnvDurationOr("POD_ABSENT_TIMEOUT", 2*time.Minute)
+func recentlyDeleted(node *wfv1.NodeStatus) bool {
+	return time.Since(node.FinishedAt.Time) <= envutil.LookupEnvDurationOr("RECENTLY_DELETED_POD_DURATION", 10*time.Second)
```

Comment:
If I'm not mistaken, the reason for using `startedAt` was because the `finishedAt` time may never have been recorded

Reply:
> this almost certainly breaks the part of https://github.com/argoproj/argo-workflows/pull/13454 which was to ensure 3.4->3.5 upgrade worked as hoped for in flight workflows.

Do you mean this problem https://github.com/argoproj/argo-workflows/issues/12103#issuecomment-2185487682?

Reply:
That is correct. @jswxstw do you think you would be able to make time for a chat about this issue?

Reply:
Actually this doesn't break the 3.4 -> 3.5 upgrade path, the else if check is present meaning we never falsely update the TaskCompletionStatus.

Reply:
> That is correct. @jswxstw do you think you would be able to make time for a chat about this issue?

Sure, I have time now.

> Actually this doesn't break the 3.4 -> 3.5 upgrade path, the else if check is present meaning we never falsely update the TaskCompletionStatus.

Yes, I kept this fix.

Reply:
> Sure, I have time now.

Its alright, I have enough confidence now that this is a better approach than #13537

Reply:
Related: https://github.com/argoproj/argo-workflows/pull/13798#discussion_r1818921785